spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -626,23 +626,31 @@ public boolean removePostProcessor(ProducerPostProcessor<K, V> postProcessor) {
 
 	@Override
 	public void updateConfigs(Map<String, Object> updates) {
-		updates.entrySet().forEach(entry -> {
-			if (entry.getKey().equals(ProducerConfig.TRANSACTIONAL_ID_CONFIG)) {
-				Assert.isTrue(entry.getValue() instanceof String, () -> "'" + ProducerConfig.TRANSACTIONAL_ID_CONFIG
-						+ "' must be a String, not a " + entry.getClass().getName());
-				Assert.isTrue(this.transactionIdPrefix != null
-								? entry.getValue() != null
-								: entry.getValue() == null,
-						"Cannot change transactional capability");
-				this.transactionIdPrefix = (String) entry.getValue();
+		updates.forEach((key, value) -> {
+			if (key == null) {
+				return;
```

Comment:
Doesn't this mean that we are going to exit from a loop on a first key as null?
Therefore we just don't examine all the entries in the map.